src/main/ruby/truffleruby/core/posix.rb

```diff
@@ -425,11 +425,62 @@ def self.read_string_native(io, length)
     end
   end
 
-  def self.read_to_buffer_native(io, length)
+  def self.execute_posix_read(io, buffer, length)
+    fd = io.fileno
+    return Truffle::POSIX.read(fd, buffer, length) unless io.timeout
+
+    deadline = Process.clock_gettime(Process::CLOCK_MONOTONIC) + io.timeout
+
+    loop do
```

Comment:
`while true` is cheaper than `loop` (which involves a block call per iteration) at least in interpreter

Reply:
TIL

Reply:
Interesting - wonder why loop is not implemented as `while true {}` than.

Reply:
Kernel#loop takes a block, it's not a keyword